library/aws/autoscale-deployment.ts

```diff
@@ -780,6 +780,20 @@ export interface AutoscaleFrontendParams extends AutoscaleProcessorParams {
    */
   customize_lb?: Customize<AR.LbParams>;
 
+  /**
+   * If true `tfgen.scopedName(name).join('-')` is used as the name.
+   * Else name is undefined and Terraform will assign a random, unique name.
+   *
+   * Note changing this for existing infra will forces new resource.
+   * see
+   * https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_target_group#name
+   *
+   * This caused a PLP prod outage.
+   *
+   * Recommend using false for existing infra and true for new.
+   */
+  target_group_generate_name: boolean;
```

Comment:
You can make it optional to prevent API breaking.

Reply:
Let's keep it explicit.